src/graph/commands/survival/survival.pas

```diff
@@ -3,16 +3,67 @@
 {$codepage UTF-8}
```

Comment:
In general this unit has become too large for good coding practices. Consider splitting some of the functionality into additional units each responsible for distinct parts of the task.